k8s/src/main/scala/io/buoyant/k8s/resources.scala

```diff
@@ -231,11 +234,15 @@ private[k8s] class NsObjectResource[O <: KubeObject: TypeReference, W <: Watch[O
       labelSelector = labelSelector,
       fieldSelector = fieldSelector,
       retryIndefinitely = true
-    ).map { maybeObj =>
-      val watch = maybeObj.toSeq.flatMap { obj => Seq(od.toWatch(obj)) }
-      // object watches don't supply a resource version, since the version associated
-      // with the object is tied to when the object was last modified, and therefore
-      // it may be too old to successfully establish a watch
-      (watch, None)
+    ).map {
+      case Some(nsObjects) =>
+        val watch = Seq(nsObjects).flatMap { obj => Seq(od.toWatch(obj)) }
+        // object watches don't supply a resource version, since the version associated
+        // with the object is tied to when the object was last modified, and therefore
+        // it may be too old to successfully establish a watch
+        (watch, None)
+      case None =>
+        log.debug(s"Initial request for api path $watchPath resulted in empty response")
+        (Seq.empty, None)
```

Comment:
Isn't this already the behavior?  `None.toSeq.flatMap{ ... }` will be `Seq.empty`.  If you just want to add the logging, you can hide it behind an `if (watch.isEmpty)`